tests/test_datasets.py

```diff
@@ -47,6 +47,101 @@ def cwd(path):
         os.chdir(oldpwd)
 
 
+def get_size(start_path="."):
+    total_byte_size = 0
+    for dir_path, dirnames, filenames in os.walk(start_path):
+        for filename in filenames:
+            # skip if lock, success, or avro file
+            if "lock" not in filename and "success" not in filename and "avro" not in filename:
+                file_path = os.path.join(dir_path, filename)
+                # skip any symlinks (not in wicker but still test)
+                if not os.path.islink(file_path):
+                    total_byte_size += os.path.getsize(file_path)
+
+    return total_byte_size
+
+
+class TestFileSystemDataset(unittest.TestCase):
+    @contextmanager
+    def _setup_storage(self) -> Iterator[Tuple[FileSystemDataStorage, WickerPathFactory, str]]:
+        with tempfile.TemporaryDirectory() as tmpdir, cwd(tmpdir):
+            fake_local_fs_storage = FileSystemDataStorage()
+            fake_local_path_factory = WickerPathFactory(root_path=tmpdir)
+            with ColumnBytesFileWriter(
+                storage=fake_local_fs_storage,
+                path_factory=fake_local_path_factory,
+                target_file_rowgroup_size=10,
+            ) as writer:
+                locs = [
+                    writer.add("np_arr", FAKE_NUMPY_CODEC.validate_and_encode_object(data["np_arr"]))  # type: ignore
+                    for data in FAKE_DATA
+                ]
+
+            arrow_metadata_table = pa.Table.from_pydict(
+                {"foo": [data["foo"] for data in FAKE_DATA], "np_arr": [loc.to_bytes() for loc in locs]}
+            )
+            metadata_table_path = os.path.join(
+                tmpdir, fake_local_path_factory._get_dataset_partition_path(FAKE_DATASET_PARTITION)
+            )
+            os.makedirs(os.path.dirname(metadata_table_path), exist_ok=True)
+            papq.write_table(arrow_metadata_table, metadata_table_path)
+
+            fake_local_fs_storage.put_object(
+                serialization.dumps(FAKE_SCHEMA).encode("utf-8"),
+                fake_local_path_factory._get_dataset_schema_path(FAKE_DATASET_ID),
+            )
+            yield fake_local_fs_storage, fake_local_path_factory, tmpdir
+
+    def test_dataset_no_cache(self):
+        with self._setup_storage() as (fake_local_storage, fake_local_path_factory, tmpdir):
+            with tempfile.TemporaryDirectory() as tmp_cache_dir:
```

Comment:
@pickles-bread-and-butter in this line above ^^^^ I don't see where `tmp_cache_dir` is passed to the `FileSystemDataset` (i.e. so when you check the size of it at the end of the test - it hasn't been used).

Reply:
It's used on line 115 in an indirect way. I meant it as the dir that would have been used as the cache to show the diff between this and the cached test. I need some dir to check for size and pretend would have been used and it seemed logical given that the other uses this variable and has a size so you can look at the diff between the two and validate it's working.

Reply:
@pickles-bread-and-butter in this line above don't you want to either change `with tempfile.TemporaryDirectory() as tmp_cache_dir:` to `with tempfile.TemporaryDirectory() as tmp_cache_dir, cwd(tmp_cache_dir):` similar to https://github.com/woven-planet/wicker/blob/ab3bbb8da0a6cbae433a1d98a7a01243ccd80b7e/tests/test_datasets.py#L67C47-L67C53

OR change the `tmpdir_size = get_size(tmp_cache_dir)` check on line 115 to be `tmpdir_size = get_size(tmdir)`?

Can you take a second look at this? It really looks like your test is equivalent to the following:
```
with tempfile.TemporaryDirectory() as tmp_cache_dir:
   # do other stuff here that uses tmpdir but not tmp_cache_dir
    tmpdir_size = get_size(tmp_cache_dir)
    assert tmpdir_size == 0
```
which of course that will pass.

Reply:
Right and that's I guess the point. Very poorly worded on my part, what I mean is that the dir that is selected as the cache in the other test is empty in this one. So looking at the two is more or less a comparison between specifying that dir as a cache or not.

So an example (maybe that makes more sense to explain my thoughts)

When you run `pytest tests/test_datasets.py` and that test is called the objective of the test is to assert that nothing is saved to the cache and you can still iterate through the data.

When the `test_dataset_cached` is called your testing if the files were moved over to this tmp_cache_dir and the data can be read correctly.

What I'm trying to convery here is what you pointed, that test will of course pass because it was never passed to the object. So the implication that the only difference between the two tests is that dir is not passed. That on a meta level tests that when you don't pass that as the cache location to the class the data is not copied.

Sorry this is a super meta test that I thought would be more clear. What you're getting at is exactly what I wanted to test but it's not directly observable unless you look at the two tests together. Does that make sense? Or do you think that it's too meta and complicated? Or maybe I'm doing something wrong I don't see.

Counter point: When you swap out the tmp-cache-dir with tmpdir in the get size call ala `tmpdir_size = get_size(tmpdir)` you're testing instead that the dir where the data written to has a certain size. That's not what we want here, we want to test that the cache we would have used wasn't written to.

Reply:
Right and that's I guess the point. Very poorly worded on my part, what I mean is that the dir that is selected as the cache in the other test is empty in this one. So looking at the two is more or less a comparison between specifying that dir as a cache or not.

So an example (maybe that makes more sense to explain my thoughts)

When you run `pytest tests/test_datasets.py` and that test is called the objective of the test is to assert that nothing is saved to the cache and you can still iterate through the data.

When the `test_dataset_cached` is called your testing if the files were moved over to this tmp_cache_dir and the data can be read correctly.

What I'm trying to convery here is what you pointed, that test will of course pass because it was never passed to the object. So the implication that the only difference between the two tests is that dir is not passed. That on a meta level tests that when you don't pass that as the cache location to the class the data is not copied.

Sorry this is a super meta test that I thought would be more clear. What you're getting at is exactly what I wanted to test but it's not directly observable unless you look at the two tests together. Does that make sense? Or do you think that it's too meta and complicated? Or maybe I'm doing something wrong I don't see.

Counter point: When you swap out the tmp-cache-dir with tmpdir in the get size call ala `tmpdir_size = get_size(tmpdir)` you're testing instead that the dir where the data written to has a certain size. That's not what we want here, we want to test that the cache we would have used wasn't written to.

Reply:
@convexquad I removed the test lol, I agree it's too much implied logic and not helpful